modules/internal/DefaultSparse.chpl

```diff
@@ -235,7 +235,114 @@ module DefaultSparse {
         add_help(ind);
       }
     }
-  
+
+    proc bulkAdd(inds: [] index(rank, idxType), sorted=false, noDuplicate=false){
+      var numAdded = inds.size; //maybe remove this var
+
+      //find individual insert points
+      //and eliminate duplicates between inds and dom
+      var indivInsertPts: [{0..#numAdded}] int;
+      var actualInsertPts: [{0..#numAdded}] int; //where to put in newdom
+
+      if !sorted then QuickSort(inds);
+
+      //eliminate duplicates --assumes sorted
+      if !noDuplicate {
+        //make sure lastInd != inds[inds.domain.low]
+        var lastInd = inds[inds.domain.low] + 1; 
+        for (i, p) in zip(inds, indivInsertPts)  {
+          if i == lastInd then p = -1;
```

Comment:
It seems like it. Do you think it worth changing? I can also define a local constant like a `NOOP_MARKER` that is -1, if you think it would clear the usage a bit more?